kotlinx-coroutines-core/jvm/src/ThreadContextElement.kt

```diff
@@ -137,6 +147,16 @@ public interface CopyableThreadContextElement<S> : ThreadContextElement<S> {
      * [CopyableThreadContextElement], implementations are performance-sensitive.
      */
     public fun copyForChildCoroutine(): CopyableThreadContextElement<S>
+
+    /**
+     * Returns a [CopyableThreadContextElement] to replace `this` `CopyableThreadContextElement` in the child
+     * coroutine's context when the child coroutine's context contains an element with the same [key] as the
+     * current one.
+     *
+     * This function is called on current element, supplied with an element retrieved from child's
+     * coroutine context by the current [key].
+     */
+    public fun merge(overwritingElement: CoroutineContext.Element): CoroutineContext
```

Comment:
Even though `CopyableThreadContextElement` is currently experimental and breaking changes are expected, it seems polite and fitting to add a default implementation for `merge` that simply returns `overwritingElement`.